File: homework_four_dot_one/src/main/java/com/abra/homework_four_dot_one/ContactItem.java
package com.abra.homework_four_dot_one;

import android.os.Parcel;
import android.os.Parcelable;

public class ContactItem implements Parcelable {
    private final int iconId;
    private final String name;
    private final String info;
    private final String typeInfo;

    public ContactItem(int iconId, String name, String info, String typeInfo) {
        this.iconId = iconId;
        this.name = name;
        this.info = info;
        this.typeInfo = typeInfo;
    }

    public int getIconId() {
        return iconId;
    }

    public String getName() {
        return name;
    }

    public String getInfo() {
        return info;
    }

    public String getTypeInfo() {
        return typeInfo;
    }

    public static final Creator<ContactItem> CREATOR = new Creator<ContactItem>() {
        @Override
        public ContactItem createFromParcel(Parcel parcel) {
            return new ContactItem(parcel);
        }

        @Override
        public ContactItem[] newArray(int i) {
            return new ContactItem[i];
        }
    };

    @Override
    public int describeContents() {
        return 0;
    }

    @Override
    public void writeToParcel(Parcel parcel, int i) {
        parcel.writeInt(iconId);
        parcel.writeString(name);
        parcel.writeString(info);
        parcel.writeString(typeInfo);
    }

    private ContactItem(Parcel parcel) {
        iconId = parcel.readInt();
        name = parcel.readString();
        info = parcel.readString();
        typeInfo = parcel.readString();
    }
}

File: homework_8_1/src/main/java/com/abra/homework_8_1/ContactItem.kt
package com.abra.homework_8_1

import android.os.Parcel
import android.os.Parcelable

class ContactItem(val iconId: Int,
                  val name: String,
                  val info: String,
                  val typeInfo: String) : Parcelable {
    constructor(parcel: Parcel) : this(
            parcel.readInt(),
            parcel.readString().toString(),
            parcel.readString().toString(),
            parcel.readString().toString())

    override fun describeContents() = 0

    override fun writeToParcel(p0: Parcel, p1: Int) {
        p0.writeInt(iconId)
        p0.writeString(name)
        p0.writeString(info)
        p0.writeString(typeInfo)
    }

    companion object CREATOR : Parcelable.Creator<ContactItem> {
        override fun createFromParcel(parcel: Parcel) = ContactItem(parcel)

        override fun newArray(size: Int): Array<ContactItem?> = arrayOfNulls(size)
    }
}